authentication/src/main/java/uk/gov/android/authentication/LoginSession.kt

```diff
@@ -0,0 +1,100 @@
+package uk.gov.android.authentication
+
+import android.app.Activity
+import android.content.Context
+import android.content.Intent
+import androidx.core.app.ActivityCompat
+import net.openid.appauth.AuthorizationException
+import net.openid.appauth.AuthorizationRequest
+import net.openid.appauth.AuthorizationResponse
+import net.openid.appauth.AuthorizationService
+import net.openid.appauth.AuthorizationServiceConfiguration
+import net.openid.appauth.TokenResponse
+import java.util.UUID
+
+class LoginSession : ILoginSession {
+    private var context: Context? = null
+    private lateinit var authService: AuthorizationService
+
+    override fun init(
+        context: Context
+    ): ILoginSession {
+        if (this.context == null) {
+            this.context = context
+            authService = AuthorizationService(context)
+        }
+        return this
+    }
+
+    override fun present(
+        configuration: LoginSessionConfiguration
+    ) {
+        if (context == null) {
+            throw Error("Context is null, did you call init?")
+        }
+
+        with(configuration) {
+            val context = this@LoginSession.context!!
+            val nonce = UUID.randomUUID().toString()
+
+            val serviceConfig = AuthorizationServiceConfiguration(
+                authorizeEndpoint,
+                tokenEndpoint
+            )
+
+            val builder = AuthorizationRequest.Builder(
+                serviceConfig,
+                clientId,
+                responseType,
+                redirectUri
+            ).also {
+                it.apply {
+                    setScopes(scopes)
+                    setUiLocales(locale)
+                    setNonce(nonce)
+                    setAdditionalParameters(
+                        mapOf(
+                            "vtr" to vectorsOfTrust
+                        )
+                    )
+                }
+            }
+
+            val authRequest = builder.build()
+
+            val authIntent = authService.getAuthorizationRequestIntent(authRequest)
+            ActivityCompat.startActivityForResult(
+                context as Activity,
+                authIntent,
+                REQUEST_CODE_AUTH,
+                null
+            )
```

Comment:
What do we use `ActivityCompat.startActivityForResult` for? I ask because I'm wondering what would happen if the Activity state isn't being monitored somehow: https://developer.android.com/training/basics/intents/result#register

Reply:
Ah it's the `@ActivityContext` that's being used: https://developer.android.com/training/dependency-injection/hilt-android#predefined-qualifiers